pkg/controller/jobs/jobset/jobset_webhook_test.go

```diff
@@ -71,3 +81,179 @@ func TestValidateCreate(t *testing.T) {
+					Labels: map[string]string{
+						constants.QueueLabel: "local-queue",
+					},
+					Namespace: "default",
+				},
+			},
+			queues: []kueue.LocalQueue{
+				*utiltesting.MakeLocalQueue("local-queue", "default").
+					ClusterQueue("cluster-queue").
+					Obj(),
+			},
+			clusterQueues: []kueue.ClusterQueue{
+				*utiltesting.MakeClusterQueue("cluster-queue").
+					AdmissionChecks("non-existent-admission-check").
+					Obj(),
+			},
+			multiKueueEnabled: true,
+			expectedManagedBy: "",
+		},
+		{
+			name: "TestDefault_MultiKueueFeatureDisabled",
+			jobSet: &jobset.JobSet{
+				ObjectMeta: ctrl.ObjectMeta{
+					Labels: map[string]string{
+						constants.QueueLabel: "local-queue",
+					},
+					Namespace: "default",
+				},
+			},
+			queues: []kueue.LocalQueue{
+				*utiltesting.MakeLocalQueue("local-queue", "default").
+					ClusterQueue("cluster-queue").
+					Obj(),
+			},
+			clusterQueues: []kueue.ClusterQueue{
+				*utiltesting.MakeClusterQueue("cluster-queue").
+					AdmissionChecks("admission-check").
+					Obj(),
+			},
+			admissionCheck: utiltesting.MakeAdmissionCheck("admission-check").
+				ControllerName(multikueue.ControllerName).
+				Active(metav1.ConditionTrue).
+				Obj(),
+			multiKueueEnabled: false,
+			expectedManagedBy: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer features.SetFeatureGateDuringTest(t, features.MultiKueue, tc.multiKueueEnabled)()
+
+			ctx, _ := utiltesting.ContextWithLog(t)
+
+			clientBuilder := utiltesting.NewClientBuilder().
+				WithObjects(
+					&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}},
+				)
+			cl := clientBuilder.Build()
+			cqCache := cache.New(cl)
+			queueManager := queue.NewManager(cl, cqCache)
+
+			for _, q := range tc.queues {
+				if err := queueManager.AddLocalQueue(ctx, &q); err != nil {
+					t.Fatalf("Inserting queue %s/%s in manager: %v", q.Namespace, q.Name, err)
+				}
+			}
+			for _, cq := range tc.clusterQueues {
+				if err := cqCache.AddClusterQueue(ctx, &cq); err != nil {
+					t.Fatalf("Inserting clusterQueue %s in cache: %v", cq.Name, err)
+				}
+				if tc.admissionCheck != nil {
+					cqCache.AddOrUpdateAdmissionCheck(tc.admissionCheck)
+					if err := queueManager.AddClusterQueue(ctx, &cq); err != nil {
+						t.Fatalf("Inserting clusterQueue %s in manager: %v", cq.Name, err)
+					}
+				}
+			}
+			webhook := &JobSetWebhook{
+				manageJobsWithoutQueueName: false,
+				queues:                     queueManager,
+				cache:                      cqCache,
+			}
+
+			err := webhook.Default(ctx, tc.jobSet)
+			if err != nil && tc.expectedError == nil {
+				t.Errorf("Unexpected error: %v", err)
+			} else if err == nil && tc.expectedError != nil {
+				t.Errorf("Expected error %v, but got nil", tc.expectedError)
+			} else if err != nil && err.Error() != tc.expectedError.Error() {
+				t.Errorf("Expected error %v, but got %v", tc.expectedError, err)
+			}
```

Comment:
nit: Can we compare errors using `cmp`, similarly as [here](https://github.com/kubernetes-sigs/kueue/blob/0446cb5e54655d3057e2e96a50de1f9da03b0a0e/pkg/controller/jobs/job/job_webhook_test.go#L377-L379)?